client/signup/main.jsx

```diff
@@ -273,6 +274,13 @@ class Signup extends React.Component {
 				} );
 			}
 		}
+
+		//`segment` query parameter
+		const siteType = queryObject.site_type;
+		if ( 'undefined' !== typeof siteType ) {
+			debug( 'From query string: site_type = %s', siteType );
+			this.props.setSiteType( siteType );
```

Comment:
This properly sets the siteType as the selected option in the list, but although the card doesn't have it as a task, I think it'd be worth skipping the step if the query parameter is set. You can follow the [example above](https://github.com/Automattic/wp-calypso/pull/28074/files#diff-7b0ca7ad49dc313c6af0636bf22edd4eR265), but that causes a flash of the step before proceeding. 

I asked @southp about a better solution in [his similar work](https://github.com/Automattic/wp-calypso/pull/28343#pullrequestreview-172777639) - maybe y'all can touch base about that to see if there's a smoother option.

Reply:
@michaeldcain To avoid the flash, one way I'm thinking of is to use `Flows.removeStepFromFlow()`. The value of siteType is set using `setSiteType()` so this value is available in all steps. 

In client/signup/main.jsx, I see that `getFlow()` is being used in a few places. We could pass a third argument to getFlow(), which contains an array of steps to remove. Or in getFlows() function definition in flows.js, we could fetch the queryObjects and iteratively remove any steps that map to the query params. Would love to hear your take if this on the right track.